src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -229,6 +229,29 @@ struct
       let is_empty {history; _} = Queue.is_empty history
     end
 
+    (* TODO: guard against waiting for transitions that already exist in the frontier *)
```

Comment:
Can this be in it's own file and follow the APIs of an extension?

Reply:
Can this be in it's own file and follow the APIs of an extension?

Reply:
Can't we use Jiawei's change on the transition_cache to wait for a transition to get into the transition frontier? In the proposer, we can add the transition that we created into the transition cache and wait for it's ivar to be filled.

Reply:
I think we'll need to move `Unprocessed_transition_cache` to somewhere in `Transition_frontier` to make the deps work out.

Reply:
Wouldn't the `Unprocessed_transition_cache` be in the processor though? In the end, I don't think `Transition_frontier` has to know about `Unprocessed_transition_cdache`?